src/linter.js

```diff
@@ -321,8 +321,26 @@ export default class Linter {
     var ScannerClass = this.getScanner(filename);
     return this.io.getFile(filename, ScannerClass.fileResultType)
       .then((fileData) => {
-        let scanner = new ScannerClass(
-          fileData, filename, {addonMetadata: this.addonMetadata});
+        // First: check that this file is under our 2MB parsing limit. Otherwise
+        // it will be very slow and may crash the lint with an out-of-memory
+        // error.
+        let fileSize = this.io.files[filename].size !== undefined ?
```

Comment:
Whilst it's not strictly necessary I think we've generally used `typeof` for undefined equality checks.